pepr/privacy/mia.py

```diff
@@ -0,0 +1,1182 @@
+        * target_indices (list): List of indices describing which of the records
+          from data were used to train the target model(s).
+        * evaluation_indices (list): List of indices describing which of the records
+          from data are used to evaluate the attack.
+        * record_indices_per_target (numpy.ndarray): n*m array describing for all n
+          target models which m indices where used in the training.
+
+    target_models: iterable
+        List of target models which should be tested.
+
+    Attributes
+    ----------
+    attack_alias : str
+        Alias for a specific instantiation of the attack class.
+    attack_pars : dict
+        Dictionary containing all needed parameters fo the attack.
+    data : numpy.ndarray
+        Dataset with all training samples used in the given pentesting setting.
+    labels : numpy.ndarray
+        Array of all labels used in the given pentesting setting.
+    data_conf : dict
+        Dictionary describing the data configuration of the given pentesting
+        setting.
+    target_models : iterable
+        List of target models which should be tested.
+    attack_results : dict
+
+        * tp_list (numpy.ndarray): True positives per attack model and target model.
+        * fp_list (numpy.ndarray): False positives per attack model and target model.
+        * fn_list (numpy.ndarray): False negatives per attack model and target model.
+        * tn_list (numpy.ndarray): True negatives per attack model and target model.
+        * test_accuracy_list (numpy.ndarray): Evaluation accuracy on evaluation records
+          per attack model and target model.
+        * precision_list (numpy.ndarray): Attack precision per attack model and target
+          model.
+        * recall_list (numpy.ndarray): Attack recall per attack model and target model.
+        * test_accuracy (numpy.ndarray): Evaluation accuracy averaged over all attack
+          models per target model.
+        * precision (numpy.ndarray): Attack precision averaged over all attack models
+          per target model.
+        * recall (numpy.ndarray): Attack recall averaged over all attack models per
+          target model.
+        * overall_test_accuracy (float): Evaluation accuracy averaged over all target
+          models.
+        * overall_precision (float): Attack precision averaged over all target models.
+        * overall_recall (float): Attack recall averaged over all target models.
+        * shadow_train_accuracy_list (list): Accuracy on training records per shadow
+          model and target model.
+        * shadow_test_accuracy_list (list): Accuracy on evaluation records per shadow
+          model and target model.
+        * shadow_train_accuracy (float): Accuracy on train records averaged over all
+          shadow models per target model.
+        * shadow_test_accuracy (float): Accuracy on evaluation records averaged over all
+          shadow models per target model.
```

Comment:
Should we rename every variable name containing ...`test`... to something with ...`eval` or `evaluation`?

Reply:
Yeah, you're right. I mean the explanation e.g. for `test_accuracy` is literally `Evaluation accuracy` :sweat_smile: Changed in 03c76ef

Reply:
In fact, evaluation and test are also two different things :)